clients/venice-push-job/src/main/java/com/linkedin/venice/vpj/pubsub/input/PubSubPartitionSplit.java

```diff
@@ -0,0 +1,235 @@
+  private long startIndex;
+
+  /** Non-serializable helpers, reconstructed lazily when needed. */
+  private transient PubSubTopicRepository topicRepository;
+  private transient PubSubTopicPartition pubSubTopicPartition;
+  private transient PubSubPosition startPubSubPosition;
+  private transient PubSubPosition endPubSubPosition;
+
+  /** Nullary constructor for deserialization. */
+  public PubSubPartitionSplit() {
+    this.topicRepository = new PubSubTopicRepository();
+  }
+
+  public PubSubPartitionSplit(
+      PubSubTopicRepository topicRepository,
+      PubSubTopicPartition pubSubTopicPartition,
+      PubSubPosition startPubSubPosition,
+      PubSubPosition endPubSubPosition,
+      long numberOfRecords,
+      int splitIndex,
+      long startIndex) {
+
+    this.topicRepository = Objects.requireNonNull(topicRepository, "topicRepository");
+    Objects.requireNonNull(pubSubTopicPartition, "pubSubTopicPartition");
+    Objects.requireNonNull(startPubSubPosition, "startPubSubPosition");
+    Objects.requireNonNull(endPubSubPosition, "endPubSubPosition");
+
+    this.topicName = pubSubTopicPartition.getTopicName();
+    this.partitionNumber = pubSubTopicPartition.getPartitionNumber();
+
+    this.startBytes = startPubSubPosition.toWireFormatBytes();
+    this.startFactoryClass = startPubSubPosition.getFactoryClassName();
+    this.endBytes = endPubSubPosition.toWireFormatBytes();
+    this.endFactoryClass = endPubSubPosition.getFactoryClassName();
+
+    this.numberOfRecords = numberOfRecords;
+    this.splitIndex = splitIndex;
+    this.startIndex = startIndex;
+
+    // Cache transient objects for in-process use
+    this.pubSubTopicPartition = pubSubTopicPartition;
+    this.startPubSubPosition = startPubSubPosition;
+    this.endPubSubPosition = endPubSubPosition;
+  }
+
+  public PubSubTopicPartition getPubSubTopicPartition() {
+    if (pubSubTopicPartition == null) {
+      ensureRepo();
+      pubSubTopicPartition = new PubSubTopicPartitionImpl(topicRepository.getTopic(topicName), partitionNumber);
+    }
+    return pubSubTopicPartition;
+  }
+
+  public PubSubPosition getStartPubSubPosition() {
+    if (startPubSubPosition == null) {
+      startPubSubPosition = PubSubPositionDeserializer.deserializePubSubPosition(startBytes, startFactoryClass);
+    }
+    return startPubSubPosition;
+  }
+
+  public PubSubPosition getEndPubSubPosition() {
+    if (endPubSubPosition == null) {
+      endPubSubPosition = PubSubPositionDeserializer.deserializePubSubPosition(endBytes, endFactoryClass);
+    }
+    return endPubSubPosition;
+  }
+
+  public String getTopicName() {
+    return topicName;
+  }
+
+  public int getPartitionNumber() {
+    return partitionNumber;
+  }
+
+  public long getNumberOfRecords() {
+    return numberOfRecords;
+  }
+
+  public int getSplitIndex() {
+    return splitIndex;
+  }
+
+  public long getStartIndex() {
+    return startIndex;
+  }
+
+  private void ensureRepo() {
+    if (topicRepository == null) {
```

Comment:
Hmm, you already have: "    this.topicRepository = Objects.requireNonNull(topicRepository, "topicRepository");" this seems unnecessary

Reply:
yeah and we create it in nullary constructor as well. Removed